### PR TITLE
Small bugs and improvement for self, self_c, contiguous()

### PR DESCRIPTION
**Pull request by sukamenev**
Please see my edits. I hope I understood your ideas correctly.

### CODE REVIEW

File: src/pointwise_ops.cpp [hunk 1]
```diff
@@ -632,8 +636,9 @@ using c10::DeviceType;
     Tensor & hardswish_(Tensor & self)
     {
         GUARD;
-        Tensor self_c = self.contiguous();
-        dlprim::Tensor x=todp(self_c);
+        if(!self.is_contiguous())
+          self = self.contiguous();
```
Comment (artyom-beilis) [inline src/pointwise_ops.cpp:640]: Same as above copy at the end

**Comment by artyom-beilis** [comment]
I added few comments. Generally if there is a modifiable or `out` tensor that isn't contiguous you work on copy and copy one back like there:

https://github.com/artyom-beilis/pytorch_dlprim/blob/34e94ef887de782061a644c1408c9eeb16f2c727/src/pointwise_ops.cpp#L713

**Comment by sukamenev** [comment]
I created new pull request with new fixes.

**Comment by sukamenev** [comment]
I created new pull request with new fixes.

**Pull request closed without merging**